src/colvar.cpp

```diff
@@ -1453,77 +1453,84 @@ cvm::real colvar::update_forces_energy()
+      // f: - initially, external biasing force
+      //    - after this code block, colvar force to be applied to atomic coordinates
+      //      ie. spring force (fb_actual will be added just below)
+      fr    = f;
+      // External force has been scaled for a 1-timestep impulse, scale it back because we will
+      // integrate it with the colvar's own timestep factor
+      f_ext = f / cvm::real(time_step_factor);
+      f_ext += (-0.5 * ext_force_k) * this->dist2_lgrad(xr, x);
+      f      = (-0.5 * ext_force_k) * this->dist2_rgrad(xr, x);
+      // Coupling force is a slow force, to be applied to atomic coords impulse-style
+      f *= cvm::real(time_step_factor);
+
+      if (is_enabled(f_cv_subtract_applied_force)) {
+        // Report a "system" force without the biases on this colvar
+        // that is, just the spring force
+        ft_reported = (-0.5 * ext_force_k) * this->dist2_lgrad(xr, x);
+      } else {
+        // The total force acting on the extended variable is f_ext
+        // This will be used in the next timestep
+        ft_reported = f_ext;
+      }
+
+      // backup in case we need to revert this integration timestep
+      // if the same MD timestep is re-run
+      prev_xr = xr;
+      prev_vr = vr;
+
+      // leapfrog: starting from x_i, f_i, v_(i-1/2)
+      vr  += (0.5 * dt) * f_ext / ext_mass;
+      // Because of leapfrog, kinetic energy at time i is approximate
+      kinetic_energy = 0.5 * ext_mass * vr * vr;
+      potential_energy = 0.5 * ext_force_k * this->dist2(xr, x);
+      // leap to v_(i+1/2)
+      if (is_enabled(f_cv_Langevin)) {
+        vr -= dt * ext_gamma * vr;
+        colvarvalue rnd(x);
+        rnd.set_random();
+        vr += dt * ext_sigma * rnd / ext_mass;
+      }
+      vr  += (0.5 * dt) * f_ext / ext_mass;
+      xr  += dt * vr;
+      xr.apply_constraints();
+      this->wrap(xr);
     } else {
-      // The total force acting on the extended variable is f_ext
-      // This will be used in the next timestep
-      ft_reported = f_ext;
-    }
-
-    // backup in case we need to revert this integration timestep
-    // if the same MD timestep is re-run
-    prev_xr = xr;
-    prev_vr = vr;
-
-    // leapfrog: starting from x_i, f_i, v_(i-1/2)
-    vr  += (0.5 * dt) * f_ext / ext_mass;
-    // Because of leapfrog, kinetic energy at time i is approximate
-    kinetic_energy = 0.5 * ext_mass * vr * vr;
-    potential_energy = 0.5 * ext_force_k * this->dist2(xr, x);
-    // leap to v_(i+1/2)
-    if (is_enabled(f_cv_Langevin)) {
-      vr -= dt * ext_gamma * vr;
-      colvarvalue rnd(x);
-      rnd.set_random();
-      vr += dt * ext_sigma * rnd / ext_mass;
-    }
-    vr  += (0.5 * dt) * f_ext / ext_mass;
-    xr  += dt * vr;
-    xr.apply_constraints();
-    this->wrap(xr);
+      // If this is a postprocessing run (eg. in VMD), the extended DOF
+      // is equal to the actual coordinate
+      xr = x;
```

Comment:
They shouldn't, it should translate to zero forces. More generally, the eABF estimators should not run in a VMD session... That is probably something to look into: ensure that ABF and metadynamics have a sane behavior when not actually running MD. I think they could be disabled entirely - the actual use cases for ABF in VMD would be very strange.

Reply:
In general, adaptive biases will have no effect on the trajectory if there is no dynamics.  But it's convenient, for example in reweighting/analysis, to get the bias energy or PMF value for a given configuration in CV space without having to look up that configuration on the PMF file.  Since you added run-time integration for ABF in higher than 1D, this is not limited to metadynamics.

Reply:
Ah yes, that would be nice. It still needs to be implemented in ABF: at this point ABF never looks up the local value of the free energy - except some variants I'm working on, and even then they don't display it. But it could populate the variable bias_energy. There is a cost to that though, which is why I didn't do it so far.